src/app/services/selection.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+
+import { Section } from '../course-list/section/section';
+import { Course } from '../course-list/course/course';
+
+@Injectable()
+export class SelectionService {
+
+  public toggleSection(section : Section) {
+    this.isSectionSelected(section) ? this.removeSection(section) : this.addSection(section);
+  }
+
+  public addSection(section : Section) {
+    let store = JSON.parse(localStorage.getItem('selections')) || {};
+    if (store[section.course_id]) {
```

Comment:
Try to avoid using nested conditionals where you can, they are harder to read. something like
```
store[section.course_id] = store[section.course_id] || [];
if (!store[section.course_id].includes(section.id)) {
  store[section.course_id].push(section.id);
  store[section.course_id].sort();
}
```
is a bit easier to read